srml/support/src/storage/hashed/generator.rs

```diff
@@ -203,15 +208,45 @@ pub trait StorageValue<T: codec::Codec> {
 	///
 	/// `T` is required to implement `codec::EncodeAppend`.
 	fn append<S: HashedStorage<Twox128>, I: codec::Encode>(
-		items: &[I], storage: &mut S
-	) -> Result<(), &'static str> where T: codec::EncodeAppend<Item=I> {
+		items: &[I],
+		storage: &mut S,
+	) -> Result<(), &'static str>
+		where T: codec::EncodeAppend<Item=I>
+	{
 		let new_val = <T as codec::EncodeAppend>::append(
 			storage.get_raw(Self::key()).unwrap_or_default(),
 			items,
 		).map_err(|_| "Could not append given item")?;
 		storage.put_raw(Self::key(), &new_val);
 		Ok(())
 	}
+
+	/// Safely append the given items to the value in the storage.
+	///
+	/// `T` is required to implement `codec::EncodeAppend`.
+	fn safe_append<'a, S: HashedStorage<Twox128>, I>(
```

Comment:
at some point if we want to make api coherent I think in the future we could introduce safe_get and get the same as append. safe_get would return default on codec error, get would return err